presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -571,11 +572,13 @@ protected Type visitGenericLiteral(GenericLiteral node, AnalysisContext context)
                 throw new SemanticException(TYPE_MISMATCH, node, "Unknown type: " + node.getType());
             }
 
-            try {
-                functionRegistry.getCoercion(VARCHAR, type);
-            }
-            catch (IllegalArgumentException e) {
-                throw new SemanticException(TYPE_MISMATCH, node, "No literal form for type %s", type);
+            if (type != JSON) {
```

Comment:
you need to use `.equals()` for types. Also, does the else branch need to provide the `json_parse` function?